src/parsing/parsing.ts

```diff
@@ -84,18 +84,49 @@ export async function resolveFiles(obj: any, currentDir: string): Promise<any> {
   return obj;
 }
 
-export function replaceTokens(obj: any, vs: Record<string, string>): any {
+export function replaceTokens(obj: any, vs: Record<string, string>, ignoreMissingVariables = false): any {
   if (Object.prototype.toString.call(obj) === '[object Object]') {
-    return mapRecord(obj, v => replaceTokens(v, vs));
+    return mapRecord(obj, v => replaceTokens(v, vs, ignoreMissingVariables));
   } else if (Array.isArray(obj)) {
-    return obj.map(element => replaceTokens(element, vs));
+    return obj.map(element => replaceTokens(element, vs, ignoreMissingVariables));
   } else if (typeof obj === 'string') {
-    return obj.replace(/%{[\w\-]+}%/g, (varName: string) => {
-      varName = varName.substr(2, varName.length - 4);
-      let v = vs[varName];
-      if (typeof v !== 'string') throw new Error(`could not find variable '${varName}'`);
-      return v;
-    });
+    const matches = obj.match(/%{([\w-]+)\s?(?:\|\|)?\s?([\w-]+)?}%/g);
```

Comment:
not sure this is the best use of groups, would this match `%{lol moon}%` ?

Reply:
ah, yes